test_conformance/relationals/test_shuffles.cpp

```diff
@@ -216,9 +216,6 @@ const char *get_order_string( ShuffleOrder &order, size_t vecSize, cl_uint lengt
     if (!byNumber) {
         byNumber = (useNumbersFlip++)%2;
     }
-    // Do not use xyzw for vectors whose length is not 2 or 4 per the spec.
-    if (vecSize != 2 || vecSize != 4 || vecSize != 3)
-        byNumber = 1;
 
     if( byNumber || vecSize > 4 )
```

Comment:
I think if we really want to test all vector sizes then we should remove this `vecSize > 4` condition also.  We can do this safely because we've already checked for indices greater than three, above.

(Aside: you didn't break this either, but it would be nice to remove the incorrect second "randomly choose upper and lower case" comment below, because the .xyzw selection is exclusively lower-case.)

Reply:
> I think if we really want to test all vector sizes then we should remove this vecSize > 4 condition also.

Yes that seems safe; I've made the change and tested it.

> (Aside: you didn't break this either, but it would be nice to remove the incorrect second "randomly choose upper and lower case" comment below, because the .xyzw selection is exclusively lower-case.)

Yes I noticed that misleading comment too and already have a followup change to tidy up the comments in this function.  I'll push that after this PR has landed.